hs_core/tests/api/rest/test_unzip.py

```diff
@@ -53,26 +53,25 @@ def setUp(self):
 
         # put the file 'test.zip' into folder 'foo'
         url4 = str.format('/hsapi/resource/{}/files/foo/', self.pid)
-        params = {'file': (payload,)}
+        params = {'file': ('test.zip',
+                           open(zip_path, 'rb'),
+                           'application/zip')}
         self.client.post(url4, params)
 
     def test_unzip(self):
         unzip_url = "/hsapi/resource/%s/functions/unzip/test.zip/" % self.pid
         response = self.client.post(unzip_url, data={"remove_original_zip": "false"})
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
+        # extra folder named as test should not exist
         list_url = "/hsapi/resource/%s/folders/test/" % self.pid
         response = self.client.get(list_url, data={})
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-
-        unzip_url = "/hsapi/resource/%s/functions/unzip/test.zip/" % self.pid
-        response = self.client.post(unzip_url, data={})
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```

Comment:
Is this actually the desired behavior?
I got the impression reading issue https://github.com/hydroshare/hydroshare/issues/4152 that we wanted to avoid the nested files like `test/test` but that we still want to retain the `test/` first directory
Maybe I missed something in conversation...

Reply:
@devincowan The issue #4152 is about not creating an extra directory using the name of the zip file without the extension .zip, so for this test case, what it used to work is to put all unzipped files into an extra ```test``` directory, hence the test asserts an extra ```test``` directory exists. If you unzip the same zip file again, it will create an extra ```test-1```, so there would be no need to check for same files names for override, since files are unzipped into a unique folder which guarantees override will never happen. My work to address this issue is to remove the extra folder, in this test case, no ```test``` extra folder will be created, instead, files will be unzipped directly to the folder where the zip file resides. If the zip file contains folders, those folders will remain as part of unzipping process, but no extra ```test``` folder will be created to put unzipped files into. Hope this explanation makes sense, but let me know if it is still confusing.

Reply:
okay I didn't know that we wanted "files will be unzipped directly to the folder where the zip file resides"
So when unzipping in-place a file `/devin.zip` I thought we wanted it to result in `/devin/*contents*` and the issue was that it previously resulted in `/devin/devin/*contents*`

Okay I'm on board now! Thanks